// node-mailwizz/dist/Countries.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Request_1 = __importDefault(require("./Request"));
const path = "/countries";
class Countries extends Request_1.default {
    /**
     * @description Creates an instance of Countries.
     * @see https://api-docs.mailwizz.com/#countries
     * @memberof Countries
     */
    constructor({ publicKey, secret, baseUrl }) {
        super({
            publicKey: publicKey,
            secret: secret,
            baseUrl: baseUrl
        });
    }
    /**
     * @description Get all countries
     * @param {GetAllCountriesParams} params - Params of the request
     * @param {number} params.page - Page of the response
     * @param {number} params.per_page - Items per page of the response
     * @returns {Promise<GetAllCountriesResponse>} - Promise of the response
     * @see https://api-docs.mailwizz.com/#get-all-countries
     * @memberof Countries
     */
    getAll({ page, per_page }) {
        this.method = Request_1.default.Type.GET;
        this.url = path;
        this.data = {
            page: page,
            per_page: per_page
        };
        return this.send();
    }
    /**
     * @description Get all zones of a country
     * @param {GetAllZonesParams} params - Params of the request
     * @param {string} params.countryID - Country ID
     * @param {number} params.page - Page of the response
     * @param {number} params.per_page - Items per page of the response
     * @returns {Promise<GetAllZonesResponse>} - Promise of the response
     * @see https://api-docs.mailwizz.com/#get-all-zones-of-a-country
     * @memberof Countries
     */
    getAllZones({ countryID, page, per_page }) {
        this.method = Request_1.default.Type.GET;
        this.url = `${path}/${countryID}/zones`;
        this.data = {
            page: page,
            per_page: per_page
        };
        return this.send();
    }
}
exports.default = Countries;
