// node-mailwizz/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateCampaignType = exports.TransactionEmail = exports.Templates = exports.ListSubscribers = exports.Lists = exports.Countries = exports.Campaigns = void 0;
const Campaigns_1 = __importDefault(require("./Campaigns"));
const Countries_1 = __importDefault(require("./Countries"));
const Lists_1 = __importDefault(require("./Lists"));
const ListSubscribers_1 = __importDefault(require("./ListSubscribers"));
const Templates_1 = __importDefault(require("./Templates"));
const TransactionEmail_1 = __importDefault(require("./TransactionEmail"));
/**
 * Classes
 */
exports.Campaigns = Campaigns_1.default;
exports.Countries = Countries_1.default;
exports.Lists = Lists_1.default;
exports.ListSubscribers = ListSubscribers_1.default;
exports.Templates = Templates_1.default;
exports.TransactionEmail = TransactionEmail_1.default;
/**
 * Campaigns API types
 */
var Campaigns_2 = require("./types/Campaigns");
Object.defineProperty(exports, "CreateCampaignType", { enumerable: true, get: function () { return Campaigns_2.CreateCampaignType; } });
