// node-mailwizz/dist/Templates.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Request_1 = __importDefault(require("./Request"));
const path = "/templates";
class Templates extends Request_1.default {
    /**
     * @description Creates an instance of Templates.
     * @see https://api-docs.mailwizz.com/#templates
     * @memberof Templates
     */
    constructor({ publicKey, secret, baseUrl }) {
        super({
            publicKey: publicKey,
            secret: secret,
            baseUrl: baseUrl
        });
    }
    /**
     * @description Get all templates
     * @param {GetAllTemplatesParams} params - Params of the request
     * @param {number} params.page - Page of the response
     * @param {number} params.per_page - Items per page of the response
     * @returns {Promise<GetAllTemplatesResponse>} - Promise of the response
     * @memberof Templates
     * @see https://api-docs.mailwizz.com/#get-all-templates
     */
    getTemplates({ page, per_page }) {
        this.url = path;
        this.method = Request_1.default.Type.GET;
        this.data = {
            page: page,
            per_page: per_page
        };
        return this.send();
    }
    /**
     * @description Get a template by UID
     * @param {GetTemplateParams} params - Params of the request
     * @param {string} params.templateUid - Template UID
     * @returns {Promise<GetTemplateResponse>} - Promise of the response
     * @memberof Templates
     * @see https://api-docs.mailwizz.com/#get-a-template
     */
    getTemplate({ templateUid }) {
        this.method = Request_1.default.Type.GET;
        this.url = `${path}/${templateUid}`;
        this.data = {};
        return this.send();
    }
}
exports.default = Templates;
